include/swift/IDE/CodeCompletion.h

```diff
@@ -668,13 +668,13 @@ class CodeCompletionResult {
   };
 
 private:
-  unsigned Kind : 3;
+  ResultKind Kind : 3;
   unsigned AssociatedKind : 8;
-  unsigned KnownOperatorKind : 6;
-  unsigned SemanticContext : 3;
-  unsigned Flair: 8;
-  unsigned NotRecommended : 4;
-  unsigned IsSystem : 1;
+  CodeCompletionOperatorKind KnownOperatorKind : 6;
+  SemanticContextKind SemanticContext : 3;
+  unsigned char Flair : 8;
+  NotRecommendedReason NotRecommended : 4;
```

Comment:
Do we have `static_assert`s that e.g. `NotRecommendedReason` doesn’t contain more then 2^4 elements?

Reply:
Good idea. I added a commit.